### PR TITLE
Contributing guidelines added for colab request

### PR DESCRIPTION
**Pull request by hardikdava**
# Description

It is a good idea to attach google colab link to easily reproduce the bug or expected results. Then minimal code will be shared among to all maintainers and contributor which will make process faster.

### CODE REVIEW

**Comment by SkalskiP** [comment]
@hardikdava, you are reading my mind. I was thinking about updating `CONTRIBUTING.md` today.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,14 +24,15 @@ Before you contribute a new feature, consider submitting an Issue to discuss the
...
-First, fork this repository to your own GitHub account. Create a new branch that describes your changes (i.e. `line-counter-docs`). Push your changes to the branch on your fork and then submit a pull request to this repository.
+First, fork this repository to your own GitHub account. Create a new branch that describes your changes (i.e. `line-counter-docs`). Push your changes to the branch on your fork and then submit a pull request to `develop` branch of this repository.
 
 When creating new functions, please ensure you have the following:
 
 1. Docstrings for the function and all parameters.
 2. Unit tests for the function.
 3. Examples in the documentation for the function.
 4. Created an entry in our docs to autogenerate the documentation for the function.
+5. Please share google colab with minimal code to test new feature or reproduce PR whenever it is possible. Please ensure that google colab can be accessed without any issue.
```
Comment (SkalskiP) [inline CONTRIBUTING.md:35]: Please add a point about opening PR and selecting `develop` as the target branch.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,14 +24,15 @@ Before you contribute a new feature, consider submitting an Issue to discuss the
 
 ## How to Contribute Changes
 
-First, fork this repository to your own GitHub account. Create a new branch that describes your changes (i.e. `line-counter-docs`). Push your changes to the branch on your fork and then submit a pull request to this repository.
+First, fork this repository to your own GitHub account. Create a new branch that describes your changes (i.e. `line-counter-docs`). Push your changes to the branch on your fork and then submit a pull request to `develop` branch of this repository.
```
Comment (SkalskiP) [inline CONTRIBUTING.md:27]: We could also add some instructions describing how to install `supervision` locally. We could link to our installation guide. https://roboflow.github.io/supervision/

Comment (SkalskiP) [reply CONTRIBUTING.md:27]: We could also mention that `supervision` has a backlog of features that we plan to add and from which you can choose. https://github.com/orgs/roboflow/projects/5/views/1

**Comment by SkalskiP** [comment]
@hardikdava, I added a few comments. Overall, we need to put more heart and effort into this file to make it more informative. And it is mostly on me. It would be optimal if we could just point new contributors to this file and be sure it contains all the necessary information.

**Comment by onuralpszr** [comment]
@SkalskiP I also thinking to add minimal colab template for speed up the process

**Comment by onuralpszr** [comment]
> @hardikdava, I added a few comments. Overall, we need to put more heart and effort into this file to make it more informative. And it is mostly on me. It would be optimal if we could just point new contributors to this file and be sure it contains all the necessary information.

I think we should also add "development" setup and we should also add how to use formatters and other tools we use in our repo to decrease our chore work as well.  Technically I will be more happy when we have formatters and proper github as well.

**Comment by hardikdava** [comment]
@SkalskiP @onuralpszr , yeah we should definitely add section of "before review" where we can explain formatter steps. But @SkalskiP please keep in mind that this guide is not going to be read by every new contributor.

**Comment by SkalskiP** [comment]
> I also thinking to add minimal colab template for speed up the process

@onuralpszr, that sounds like a good idea!

**Comment by SkalskiP** [comment]
> But please keep in mind that this guide is not going to be read by every new contributor.

@hardikdava agree! Still, it is better to have it ready. I think.

**Comment by mayankagarwals** [comment]
> Technically I will be more happy when we have formatters and proper github action as well.

One thing I noticed as a new contributor was failing github actions because of unformatted code. 
https://github.com/roboflow/supervision/blob/8662671e6b4a1793602e4981ba8e164a2e5d50e3/.github/workflows/test.yml#L30C16-L30C34 

Can this not be avoided by simply using black to not just check but autoformat code? 
https://github.com/roboflow/supervision/blob/8662671e6b4a1793602e4981ba8e164a2e5d50e3/Makefile#L11

**Comment by SkalskiP** [comment]
@onuralpszr can't wait for that pre-commit magic!

All of that is awesome conversation. My question now is: @hardikdava and @onuralpszr do we merge this PR for now and add all of those ideas in separate PRs?

**Comment by onuralpszr** [comment]
> @onuralpszr can't wait for that pre-commit magic!
> 
> All of that is awesome conversation. My question now is: @hardikdava and @onuralpszr do we merge this PR for now and add all of those ideas in separate PRs?

Merge this docs part so I can add my parts and we can move forward. I was already writing it. waiting for this one to be complete as well.

**Comment by SkalskiP** [comment]
Merging.

**Pull request merged**